code/datums/components/spill.dm

```diff
@@ -0,0 +1,38 @@
+// This component is for forcing strange things into your pocket that fall out if you fall down
+// Yes this exists purely for the spaghetti meme
+
+/datum/component/spill
+	var/preexisting_item_flags
+
+/datum/component/spill/Initialize()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/spill/PostTransfer()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/spill/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/equip_react)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/drop_react)
+	var/obj/item/master = parent
+	preexisting_item_flags = master.item_flags
+	master.item_flags |= ITEM_SLOT_POCKET
+
+/datum/component/spill/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED))
+	var/obj/item/master = parent
+	if(!(preexisting_item_flags & ITEM_SLOT_POCKET))
+		master.item_flags &= ~ITEM_SLOT_POCKET
+
+/datum/component/spill/proc/equip_react(obj/item/source, mob/equipper, slot)
+	if(slot == SLOT_L_STORE || slot == SLOT_R_STORE)
+		RegisterSignal(equipper, COMSIG_LIVING_STATUS_KNOCKDOWN, .proc/knockdown_react, TRUE)
+	else
+		UnregisterSignal(equipper, COMSIG_LIVING_STATUS_KNOCKDOWN)
+
+/datum/component/spill/proc/drop_react(obj/item/source, mob/dropper)
+	UnregisterSignal(dropper, COMSIG_LIVING_STATUS_KNOCKDOWN)
+
+/datum/component/spill/proc/knockdown_react(mob/living/fool)
+	fool.dropItemToGround(parent)

```

Comment:
Make a funny message about spagocki out pocketti

Reply:
I was actually just working on that